api/routes/articles.js

```diff
@@ -8,179 +8,208 @@ const pool = new Pool({
+              SELECT rid,aid 
+              FROM authorizations 
+              WHERE rid in (
+                SELECT rid 
+                FROM authorizations 
+                WHERE aid = ${aid}
+              )
+            ) as a NATURAL JOIN articles
+          ) as b
+          WHERE rank <= 3
+        ) as c NATURAL JOIN researchers
+        ORDER BY name
+      `;
+
+      const authorInfoObject = await client.query(query);
+
+      if (authorInfoObject && authorInfoObject.rowCount > 0) {
+        var curauthorName = authorInfoObject.rows[0].name;
+        var curauthor = { name: curauthorName, otherHighlyCitedArticles: [] };
+
+        authorInfoObject.rows.forEach(function (item) {
+          if (item.name == curauthorName) {
+            curauthor.otherHighlyCitedArticles.push(item.title);
+          } else {
+            articleQueryObject.authorInfo.push(curauthor);
+            curauthorName = item.name;
+            curauthor = {
+              name: item.name,
+              otherHighlyCitedArticles: [item.title]
+            };
+          }
         }
+        );
+        articleQueryObject.authorInfo.push(curauthor);
+      }
+      if (fid) {
+        query = `
+          SELECT title 
+          FROM articles 
+          WHERE fid = ${fid} 
+          ORDER BY citation DESC LIMIT 5
+        `;
+        const fieldarticles = await client.query(query);
+        fieldarticles.rows.forEach(function (item) {
+          articleQueryObject.fieldInfo.push(item.title);
+        });
       }
-    );
-    articleQueryObject.authorInfo.push(curauthor);
-    }
-  // console.log(typeof fid);
-    if (fid) {
-      //console.log('here!');
-      query = `SELECT title FROM articles WHERE fid = ${fid} ORDER BY citation DESC LIMIT 5`;
-      const fieldarticles = await client.query(query);
-      fieldarticles.rows.forEach(function(item) {
-        articleQueryObject.fieldInfo.push(item.title);
-      });
-    }
     }
+
     client.release();
-  //  console.log(articleQueryObject);
     res.send(articleQueryObject);
-  } catch(err) {
+  } catch (err) {
     console.error(err);
+    res.send('not found')
   }
 })
 
-//  POST /articles add
-router.post('/',async (req, res) => {
-
+//  POST /articles 
+router.post('/', async (req, res) => {
   const frontArticle = req.body;
   const client = await pool.connect();
+
   try {
-      const partialArticle = {title:null,authors:[],publishYear:null,url:null,citations:null,fieldID:null};
-      const keys = Object.keys(frontArticle);
-      keys.forEach(key => {
-        partialArticle[key] = frontArticle[key];
-      });
-      await client.query('BEGIN');
-      var query = `INSERT INTO articles (title,pub_year,url,citation,fid) values ('${partialArticle.title}',${partialArticle.publishYear},'${partialArticle.url}',${partialArticle.citations},'${partialArticle.fieldID}') RETURNING aid`;
-      const rinserted = await client.query(query);
-      const insertaid = rinsertid.rows[0].aid;
```

Comment:
Was this endpoint tested? There's a typo here. `rinserted` vs `rinsertid`

Reply:
No. This has not been tested. Can I modify it right here?